python/dev-requirements.txt

```diff
@@ -27,3 +27,6 @@ types-redis~=4.3.21
 types-tzlocal~=4.2
 cryptography
 jupyter
+sentence-transformers
```

Comment:
Do we still need to specify these dependencies in dev-requirement if the Notebook examples already have the code to install these dependencies?